backend/isic/isic_crawler.py

```diff
@@ -0,0 +1,98 @@
+import requests
+import pathlib
+import os
+from io import BytesIO
+from PIL import Image, ImageFile
+ImageFile.LOAD_TRUNCATED_IMAGES = True
+
+
+"""A module to fetch data from International Skin Imaging Collaboration API
+
+For further information about the API: https://isic-archive.com/api/v1/
+"""
+
+
+def ISIC_request(response, num=100):
+    """Fetching the metadata from images in ISIC archive by sending a HTTP
+    request to the ISIC's REST API.
+    Parameters:
+        num(int): Number of the images you get from the archive,default is 100
+    Returns:
+        Json: Response metadata in JSON format
+    """
+    start_url = f'https://isic-archive.com/api/v1/image?limit={num} \
+                  &sort=name&sortdir=1&detail=true'
+    headers = {'Accept': 'application/json'}
+    r = requests.get(start_url, headers=headers)
+    data = r.json()
+    return data
+
+
+def check_dermoscopic(meta):
+    """Checking if a image is acquired through dermoscopy by ISIC's API.
+    Parameter:
+    meta: The metadata of the image getting through the API
+    Return:
+    True if the image is acquired through dermoscopy, False if it isn't
+    """
+
+    if "image_type" not in meta["meta"]["acquisition"]:
+        return False
+    elif not str(meta["meta"]["acquisition"]["image_type"]) == "dermoscopic":
+        return False
+    return True
+
+
+def check_melanoma(metadata):
+    """Checking if a image is confirmed melanocytic by ISIC's API.
+    Parameter:
+        metadata: The metadata of the image getting through the API
+    Return:
+        True if the image is confirmed melanocytic, False if it isn't
+    """
+    if "melanocytic" not in metadata["meta"]["clinical"]:
+        return False
+    elif not metadata["meta"]["clinical"]["melanocytic"]:
+        return False
+    return True
+
+
+def ISIC_getdata(numjson=20, numimage=20,
```

Comment:
This function needs to check for downloaded images. If there are ALREADY downloaded image, it does NOT redownload from isic and overwrite that file.

Reply:
Yea, saving time. By the way, @casker2209 could you also sort ascending so that the first downloaded images are more lightweight?

Reply:
Yeah I did it. The API already implement ascending and descending sorting, I just had to change some minor thing and I have done it